src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/delete_command_request.cc

```diff
@@ -79,9 +79,15 @@ void DeleteCommandRequest::Run() {
   const int32_t cmd_id =
       (*message_)[strings::msg_params][strings::cmd_id].asInt();
 
-  smart_objects::SmartObject* command = application->FindCommand(cmd_id);
+  smart_objects::SmartObject command;
+  {
+    smart_objects::SmartObject* command_ptr = application->FindCommand(cmd_id);
+    if (command_ptr) {
+      command = *command_ptr;
```

Comment:
@LuxoftAKutsan Seems there is still a chance to get the command object destroyed before copying it. Is this ok?

Reply:
@LuxoftAKutsan

Agreed with @mvorobio, I think the only way to make this truly safe is to add a function which returns by value instead of pointer.